### ISSUE TITLE
Non-zero exit code breaks CI compatibility

### ISSUE THREAD

**Issue by madlabman** [protofire/solhint#572]
Since the version **4.5.0** there's a zero exit code (see #554) even with reported errors. It makes it hard to use within actions in CI. My personal opinion is to have the similar behaviour other linters preserve - return 1 in case of errors.

I assume the implemented change was an attempt to fix kinda niche issue (even it has fixed my workflow as well), and it was done in bad manner, I believe. My suggestion is to use command line flags to achieve the desired behaviour, such as **--exit-code** (to return non-zero exit code back) or, which is even better, to re-introduce the non-zero exit code and use a flag to change the behaviour on demand.

**Comment by dbale-altoros** [comment]
I will check this as soon as i can
thanks @madlabman for the suggestion

**Comment by juanpcapurro** [comment]
FWIW I overhauled the exit codes on solhint-community with:

 https://github.com/solhint-community/solhint-community/pull/134 --  to exit early and with a different code if there's a misconfiguration
 https://github.com/solhint-community/solhint-community/issues/76 -- to have --quiet not override --max-warnings

**Comment by dbale-altoros** [comment]
> FWIW I overhauled the exit codes on solhint-community with:
> 
> [solhint-community/solhint-community#134](https://github.com/solhint-community/solhint-community/pull/134) -- to exit early and with a different code if there's a misconfiguration [solhint-community/solhint-community#76](https://github.com/solhint-community/solhint-community/issues/76) -- to have --quiet not override --max-warnings

cool!
Thanks @juanpcapurro  for that!

**Comment by MaxenceAdnot** [comment]
Hello @dbale-altoros 

Do you plan to implement this in solhint or should we move to solhint-community ?

**Comment by dbale-altoros** [comment]
Hey @MaxenceAdnot I will implement this fix as well
Probably next week
solhint community is a fork from protofire solhint
both are open source

We will mantain this original solhint because it is an important part of protofire products

**Comment by dbale-altoros** [comment]
@MaxenceAdnot 
@madlabman 
Expect new solhint version on Monday

**Comment by dbale-altoros** [comment]
Will merge this
https://github.com/protofire/solhint/pull/578
To fix this issue

**Comment by dbale-altoros** [comment]
New version
https://www.npmjs.com/package/solhint

**Comment by MaxenceAdnot** [comment]
Thanks a lot @dbale-altoros

**Comment by madlabman** [comment]
![image](https://github.com/protofire/solhint/assets/10616301/1403645d-45d8-4a56-b5ae-3eaead1d66ba)
Welp, still doesn't work for me.

Probably, the reason is the following code
```js
if (reports[0].errorCount > 0) process.exit(EXIT_CODES.REPORTED_ERRORS)
```
should iterate over the all the reports instead.

@dbale-altoros

**Comment by dbale-altoros** [comment]
@madlabman 
Yes , sorry. You're right
I mean, reports[0] contains all errors from the first contract analyzed
If you have more than one contract this will not provide correct info
Like you said I need to loop through all contracts ( all reports[] )
I will fix it today on my afternoon

Thanks a lot for the feedback !! Really appreciate

**Comment by dbale-altoros** [comment]
@madlabman made a new package on npm
whenever you can, please confirm if it is working

**Comment by madlabman** [comment]
> @madlabman made a new package on npm whenever you can, please confirm if it is working

LGTM, thank you @dbale-altoros !
